README.md

```diff
@@ -1 +1,242 @@
+[![Gem Version](https://badge.fury.io/rb/simple-crud.svg)](https://badge.fury.io/rb/simple-crud)
+[![Code Climate](https://codeclimate.com/github/Wolox/simple-crud/badges/gpa.svg)](https://codeclimate.com/github/Wolox/simple-crud)
+
+# Table of contents
+  - [Description](#description)
+  - [Installation](#installation)
+  - [Usage](#usage)
+    - [Setup](#setup)
+      - [Application controller](#application-controller)
+      - [Each controller](#each-controller)
+      - [Options](#options)
+        - [Paginate](#paginate)
+        - [Authorize](#authorize)
+        - [Authenticate](#authenticate)
+        - [Serializer](#serializer)
+        - [Filter](#filter)
+      - [Shared examples](#shared-examples)
+  - [Contributing](#contributing)
+  - [Releases](#releases)
+  - [About](#about)
+  - [License](#license)
+
+-----------------------
+
+## Description
+
+Simple Crud is a gem for Rails that simplifies writing standard CRUD actions, while also adding tools so it's not needed to write tests for them. Its main objective is replacing generally copy-pasted code like generic paginated, authenticated, filtered index methods for simple lines such as
+```ruby
+simple_crud_for :index
+```
+It includes support for index, create, destroy, update and show, and options to specify whether it should apply pagination, authorization, authentication, filtering and if a particular serializer should be used. Keep in mind, though, that the idea is not to replace writing methods in controllers altogether, but only to replace most standard cases.
+
+## Installation
+Add the following line to your application's Gemfile:
+
+```ruby
+gem 'simple_crud'
+```
+
+And then execute:
+```bash
+$ bundle
+```
+
+Or install it yourself as:
+```bash
+$ gem install simple_crud
+```
+
+## Usage
+### Setup
+#### Application controller
+Before SimpleCrud can be used, some boilerplate is needed. Add the following to your ApplicationController
+
+```ruby
+include Pundit
+include Wor::Paginate
+extend SimpleCrud
+
+before_action :set_params
+
+def set_params
+  SimpleCrudController.params = params
+end
+
+rescue_from ActiveRecord::RecordNotFound, with: :not_found
+rescue_from ActiveRecord::StatementInvalid, with: :unprocessable
+rescue_from ActiveRecord::RecordInvalid, with: :unprocessable
+rescue_from ActionController::ParameterMissing, with: :unprocessable
+rescue_from Pundit::NotAuthorizedError, with: :forbidden
+
+def not_found(_exception)
+  head :not_found
+end
+
+def unprocessable(exception)
+  render json: { errors: exception }, status: :unprocessable_entity
+end
+
+def forbidden(exception)
+  render json: { errors: exception }, status: :forbidden
+end
+```
+
+### Each controller
+In case you need either update or create, create a method with the valid input params using standard rails:
+```ruby
+def author_params
+  params.permit(:first_name, :email, :last_name, :institution, :role)
+end
```

Comment:
What if I need to edit different parameters not present on create? Can I do that ?